R/utils.R

```diff
@@ -0,0 +1,114 @@
+## Microbiome-specific helpers
+
+
+## Make default tree when none supplied
+## NOT phase 1 - leaving for a later phase
+# makeDefaultTree <- function(taxonomy_df) {
+  
+#     #### Eventually needs to be only the actual var details
+#     #### Eventually needs to make the formula based on col names
+#     # phylo.formula <- ...
+#     tree <- ape::as.phylo.formula(~Kingdom/Phylum/Class/Order, tax_df)
+  
+#     nEdges <- dim(tree$edge)[1]
+    
+#     # Assign edge lengths to all equal 1. 
+#     tree$edge.length <- rep(1, nEdges)
+    
+#     # Root tree using the midpoint
+#     # tree <- phytools::pbtree(n=20,scale=30)
+#     tree <- phytools::midpoint.root(tree)
+    
+#     # Force the tree to be ultrametric (all tips equadistant from root)
+#     tree <- phytools::force.ultrametric(tree, method="extend")
+
+#     return(tree)
+# }
+
+#' @import data.table
+#' @import veupathUtils
+rankTaxa <- function(df, method=c('median','max','q3','var')) {
```

Comment:
might make sense to have multiple utils-* files. i know its early but i think we can at least anticipate json having its own set of utils